src/pudl/output/new_eia923.py

```diff
@@ -0,0 +1,222 @@
+    """A factory for aggregated nuclear and non-nuclear generation fuel assets."""
+    table_root = "generation_fuel_nuclear" if nuclear else "generation_fuel"
+    input_asset = f"{table_root}_eia923"
+    output_asset = f"{table_root}_{AGG_FREQS[freq]}_eia923"
+
+    @asset(
+        ins={input_asset: AssetIn()},
+        name=output_asset,
+        io_manager_key=io_manager_key,
+        compute_kind="Python",
+    )
+    def gen_fuel(**ins) -> pd.DataFrame:
+        gf_df = ins[input_asset]
+        # Create a date index for temporal resampling:
+        gf_df = gf_df.set_index(pd.DatetimeIndex(gf_df.report_date))
+        by = [
+            "plant_id_eia",
+            "fuel_type_code_pudl",
+            "energy_source_code",
+            "prime_mover_code",
+        ]
+        if nuclear:
+            by = by + ["nuclear_unit_id"]
+        by = by + [pd.Grouper(freq=freq)]
+
+        # Sum up these values so we can calculate quantity weighted averages
+        gf_gb = gf_df.groupby(by=by, observed=True)
+        gf_df = gf_gb.agg(
+            {
+                "fuel_consumed_units": pudl.helpers.sum_na,
+                "fuel_consumed_for_electricity_units": pudl.helpers.sum_na,
+                "fuel_consumed_mmbtu": pudl.helpers.sum_na,
+                "fuel_consumed_for_electricity_mmbtu": pudl.helpers.sum_na,
+                "net_generation_mwh": pudl.helpers.sum_na,
+            }
+        )
+        gf_df["fuel_mmbtu_per_unit"] = (
+            gf_df["fuel_consumed_mmbtu"] / gf_df["fuel_consumed_units"]
+        )
+
+        gf_df = gf_df.reset_index()
+        return gf_df
+
+    return gen_fuel
+
+
+generation_fuel_assets = [
+    generation_fuel_eia923_aggregated_asset_factory(
+        freq=freq, nuclear=nuke, io_manager_key=None
+    )
+    for freq in AGG_FREQS
+    for nuke in [True, False]
+]
+
+
+def generation_fuel_all_eia923_aggregated_asset_factory(
+    freq: Literal["AS", "MS"] | None,
+    io_manager_key: str | None = None,
+) -> AssetsDefinition:
+    """Factory for aggregated combined nuclear & non-nulcear generation fuel assets."""
+    freq_str = AGG_FREQS[freq] + "_" if freq is not None else ""
+
+    @asset(
+        ins={
+            f"generation_fuel_{freq_str}eia923": AssetIn(),
+            f"generation_fuel_nuclear_{freq_str}eia923": AssetIn(),
+        },
+        name=f"generation_fuel_all_{freq_str}eia923",
+    )
+    def gen_fuel_all(**ins) -> pd.DataFrame:
+        return pudl.output.eia923.generation_fuel_all_eia923(
+            gf=ins[f"generation_fuel_{freq_str}eia923"],
+            gfn=ins[f"generation_fuel_nuclear_{freq_str}eia923"],
+        )
+
+    return gen_fuel_all
+
+
+generation_fuel_all_assets = [
+    generation_fuel_all_eia923_aggregated_asset_factory(freq=freq, io_manager_key=None)
+    for freq in list(AGG_FREQS) + [None]
+]
+
+
+################################################################################
+# Denormalization down here. Not dealing with it for now.
```

Comment:
I separated the denormalization part because there are so many tables floating around, and I wasn't sure we wanted these IDs added to every single one of them, and also because it seemed like maybe it could just be a re-usable function rather than doing almost the same thing in every table.

Which tables should exist in normalized and denormalized forms?  Which tables are we expecting users to view and need to be readable? Should I just cut-and-paste the per-table normalization steps into every asset factory and combine them into one step, even though it's exactly the same merge regardless of whether it's an unaggregated, monthly, or yearly output?